### PR TITLE
IA-1828 Upgrade scalatest to 3.1.2

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/IA-1828

Tried scalafix for upgrading scalatest, it's pretty convenient (https://github.com/scalatest/autofix/tree/master/3.1.x#autofix-for-scalatest-31x).

See if upgrading scalatest makes a difference

Main breaking changes are:
```
import org.scalatest.FlatSpec
import org.scalatest.Matchers
```
are now
```
import org.scalatest.flatspec.AnyFlatSpec
import org.scalatest.matchers.should.Matchers
```

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by Qi77Qi** [comment]
jenkins retest

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala [hunk 1]
```diff
@@ -266,13 +265,14 @@ trait LeonardoTestUtils
...
     // verify with get()
-    val creatingCluster = eventually {
-      verifyCluster(Leonardo.cluster.get(googleProject, clusterName),
-                    googleProject,
-                    clusterName,
-                    List(ClusterStatus.Creating),
-                    clusterRequest)
-    }(getAfterCreatePatience, implicitly[Position])
+    val creatingCluster =
+      eventually(Timeout(getAfterCreatePatience.timeout), Interval(getAfterCreatePatience.interval)) {
```
Comment (Qi77Qi) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala:269]: `eventually` in new scalatest takes more implicits if we use the old format...hence using a different overloaded function that does the same thing

**Review by rtitle — APPROVED** [review]
LGTM

File: http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/dao/MockSamDAO.scala [hunk 2]
```diff
@@ -50,6 +50,8 @@ class MockSamDAO extends SamDAO[IO] {
           .map(_.contains(action)) //open the option to peek the set: Option[Bool]
           .getOrElse(false) //unpack the resulting option and handle the disk never having existed
         IO.pure(res)
+      case SamResourceType.App =>
```
Comment (Qi77Qi) [inline http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/dao/MockSamDAO.scala:53]: @jdcanas   just fyi,  I'm fixing this in this PR...it's a bit distracting seeing 3 compilation errors everytime

**Comment by Qi77Qi** [comment]
seems there's some flake in org.broadinstitute.dsde.workbench.leonardo.db.KubernetesClusterComponentSpec

jenkins retest

**Pull request merged**